e2e/inspect/inspect.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+// This test sets singularity image specific environment variables and
+// verifies that they are properly set.
+
+package singularityenv
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+
+	"github.com/buger/jsonparser"
+	"github.com/sylabs/singularity/e2e/internal/e2e"
+	"github.com/sylabs/singularity/internal/pkg/test"
+)
+
+type testingEnv struct {
+	// base env for running tests
+	CmdPath string `split_words:"true"`
+}
+
+var testenv testingEnv
+
+const expectedLabelsJson = `
+{
+	"attributes": {
+		"apps": "",
+		"labels": {
+			"E2E": "AWSOME",
+			"HI": "\"HELLO WORLD\"",
+			"e2e": "awsome",
+			"hi": "\"hello world\"",
+			"org.label-schema.build-date": "Friday_14_June_2019_11:59:57_PDT",
+			"org.label-schema.schema-version": "1.0",
+			"org.label-schema.usage": "/.singularity.d/runscript.help",
+			"org.label-schema.usage.singularity.deffile.bootstrap": "library",
+			"org.label-schema.usage.singularity.deffile.from": "alpine:latest",
+			"org.label-schema.usage.singularity.runscript.help": "/.singularity.d/runscript.help",
+			"org.label-schema.usage.singularity.version": "3.2.1-660.g4c8a84050"
+		}
+	},
+	"type": "container"
+}`
+
+const expectedRunscriptJson = `
+{
+	"attributes": {
+		"apps": "",
+		"runscript": "#!/bin/sh\n\necho \"E2E Test Container\"\n\n\n"
+	},
+	"type": "container"
+}`
+
+const expectedListAppsJson = `
+{
+	"attributes": {
+		"apps": "hello\nworld\n"
+	},
+	"type": "container"
+}`
+
+const expectedTestJson = `
+{
+	"attributes": {
+		"apps": "",
+		"test": "#!/bin/sh\n\nls /\ntest -d /\ntest -d /etc\n\n\n"
+	},
+	"type": "container"
+}`
+
+const expectedEnvironmentJson = `
+{
+	"attributes": {
+		"apps": "",
+		"environment": {
+			"90-environment.sh": "#!/bin/sh\n#Custom environment shell code should follow\n\n\nexport test=\"testing\"\nexport e2e=\"e2e testing\"\n\n\n"
+		}
+	},
+	"type": "container"
+}`
+
+func runInspectCommand(inspectType string) ([]byte, error) {
+	argv := []string{"inspect", "--json", inspectType, "testdata/test.sif"}
```

Comment:
the name "test.sif" is too generic. If this file is specific to this set of tests, its name should reflect that.

Also, if there's something special about this file, instructions as to how to recreate it should be included. Alternatively the file should be created as part of the test. Or use the standard test file provided in the environment variable E2E_IMAGE_PATH.